app/controllers/user/masquerades_controller.rb

```diff
@@ -0,0 +1,14 @@
+class User::MasqueradesController < Devise::MasqueradesController
```

Comment:
Hmmm. good point.

Here we find a user, go to their page, then click masquerade? If so, then the user namespace is better. Otherwise, use the admin namespace.